### PR TITLE
[PREVIEW] Div 2815 check payment status

### PR DESCRIPTION
**Pull request by david-gimelle**
# Description

See Ticket Div-2815

In some case the petitioner could have done a successful payment but CCD has not been updated with it. So when a petitioner login, if the petition is in state Awaiting Payment, the application will check with payment application if a success full payment has been made and update CCD.

Use this PR for testing https://github.com/hmcts/div-petitioner-frontend/pull/321 - this PR has check from ccd activated correctly

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- Submit a petition, make a payment and logout immediately.
- Manually check the payment is successful but CCD is still in Awaiting Payment
- Log in petition
- Check in CCD the State is not Awaiting Payment anymore.

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings
- [ x] I have added tests that prove my fix is effective or that my feature works
- [x ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/middleware/submissionMiddleware.test.js [hunk 1]
```diff
@@ -72,4 +75,96 @@ describe(modulePath, () => {
...
   });
+
+  describe('#paymentAwaiting', () => {
+    let getToken = null;
+    let query = null;
+    let update = null;
+    beforeEach(() => {
+      getToken = sinon.stub().resolves('token');
+      // Payment create stub
+      query = sinon.stub().resolves({
```
Comment (tchow8) [inline app/middleware/submissionMiddleware.test.js]: Both of your tests set query so I think this can be be removed, as it would be unused?

Comment (david-gimelle) [reply app/middleware/submissionMiddleware.test.js]: Done

File: app/middleware/submissionMiddleware.js [hunk 2]
```diff
@@ -1,13 +1,52 @@
...
     case 'AwaitingPayment':
+      if (session.currentPaymentReference) {
+        return paymentStatusService
+          .checkAndUpdatePaymentStatus(res, user, authToken, session)
+          .then(response => {
+            if (response !== true) return '/application-submitted-awaiting-response';
+            return '/application-submitted';
+          })
+          // Log any errors occurred and end up on the error page.
+          .catch(error => {
```
Comment (tchow8) [inline app/middleware/submissionMiddleware.js]: Can you put the catch block as the last block? Thrown errors in any of the promises the catch block is chained to will eventually reach there, so I think its cleaner to have a single catch block at the end of a promise chain.

Comment (tchow8) [reply app/middleware/submissionMiddleware.js]: And just use res.redirect('/generic-error') here. We should have a separation in the success and error flows.

Comment (david-gimelle) [reply app/middleware/submissionMiddleware.js]: Done

File: app/steps/pay/pay-online-only/index.js [hunk 3]
```diff
@@ -117,6 +117,7 @@ module.exports = class PayOnline extends Step {
         const { id, status, reference, nextUrl } = response;
         req.session.currentPaymentId = id;
         req.session.currentPaymentReference = reference;
+        req.session.payment_reference = reference;
```
Comment (tchow8) [inline app/steps/pay/pay-online-only/index.js]: Any reason we can't just use one of the two currentPAymentReference or payment_reference for everything if they are the same assignment?

**Comment by nsingamsetty** [comment]
This is tested on FE PR-https://pr-321-div-pfe-preview.service.core-compute-preview.internal/index and working as Expected.
Please find the attached evidences. All E2E automation scripts executed and passed against PR Reports Attached in the ticket for evidences. Ready to Merge this PR.Reports and Evidences can be viewed in the JIRA-2815

**Pull request merged**